homeassistant/components/lektrico/sensor.py

```diff
@@ -0,0 +1,294 @@
+"""Support for Lektrico charging station sensors."""
+from __future__ import annotations
+
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+import logging
+from typing import Any
+
+from lektricowifi import lektricowifi
+
+from homeassistant.components.sensor import (
+    DEVICE_CLASS_CURRENT,
+    DEVICE_CLASS_ENERGY,
+    DEVICE_CLASS_ILLUMINANCE,
+    DEVICE_CLASS_POWER,
+    DEVICE_CLASS_TEMPERATURE,
+    DEVICE_CLASS_VOLTAGE,
+    STATE_CLASS_MEASUREMENT,
+    STATE_CLASS_TOTAL_INCREASING,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_SW_VERSION,
+    CONF_FRIENDLY_NAME,
+    ELECTRIC_CURRENT_AMPERE,
+    ELECTRIC_POTENTIAL_VOLT,
+    ENERGY_KILO_WATT_HOUR,
+    POWER_KILO_WATT,
+    TEMP_CELSIUS,
+    TIME_SECONDS,
+)
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers import device_registry as dr, entity_registry as er
+from homeassistant.helpers.entity import DeviceInfo
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+PARALLEL_UPDATES = 1
+SCAN_INTERVAL = timedelta(seconds=10)
+
+
+@dataclass
+class LektricoSensorEntityDescription(SensorEntityDescription):
+    """A class that describes the Lektrico sensor entities."""
+
+    unit_fn: str | None = None
+    value_fn: Any | None = None
+
+
+SENSORS: tuple[LektricoSensorEntityDescription, ...] = (
+    LektricoSensorEntityDescription(
+        key="charger_state",
+        name="Charger_State",
+        value_fn=lambda x: x.charger_state,
+    ),
+    LektricoSensorEntityDescription(
```

Comment:
What does the charging time value represent? Could you describe the behavior of this sensor to me? It potentially isn't right based on what it does...

Reply:
"charging_time" is an integer, it comes from the physical device and it shows  the current session charging time in seconds.
The physical device is charging electrical vehicles.
When no one is charging, charging_time comes 0.
When a charging starts, a charging session is started and charging_time shows the time that the vehicle is charging.
The vehicle can stop the charging (and charging_time will remain the same) and the vehicle can start again to charge (and charging_time will increase).
When the owner wants to stop (disconnects the cable, or swipes a NFC tag, or gives a command in the application), the charging session is stopped => charging_time is reseted.